File: src/main/java/me/onixdev/events/bukkit/PlayerReleaseUseItemState.java
package me.onixdev.events.bukkit;

import me.onixdev.OnixAnticheat;
import me.onixdev.user.OnixUser;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerItemConsumeEvent;

import java.util.UUID;

public class PlayerReleaseUseItemState implements Listener {
    @EventHandler
    public void onConsume(PlayerItemConsumeEvent event) {
        Player player = event.getPlayer();
        UUID uuid = player.getUniqueId();
        OnixUser user = OnixAnticheat.INSTANCE.getPlayerDatamanager().get(uuid);
        if (user == null){
            return;
        }
        user.setUsingItem(false);
    }
}

File: src/main/java/me/onixdev/events/bukkit/PlayerReleaseUseItemState.kt
package me.onixdev.events.bukkit

import me.onixdev.OnixAnticheat
import me.onixdev.util.extend.KotlinExtends.getData
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerItemConsumeEvent


class PlayerReleaseUseItemState : Listener {
    @EventHandler
    fun onConsume(event: PlayerItemConsumeEvent) {
        val player = event.getPlayer()
        val user = player.getData()
        user?.setUsingItem(false)
    }
}
